packages/@expo/fingerprint/src/hash/ReactImportsPatcher.ts

```diff
@@ -0,0 +1,360 @@
+   * @default true given the first chunk is usually good enough for imports.
+   */
+  onlyTransformFirstChunk?: boolean;
+
+  /**
+   * A function that transforms a chunk of data. Exposing this for testing purposes.
+   * @default The `patchChunk`
+   */
+  transformFn?: (chunk: string) => string;
+}
+
+/**
+ * A transform stream that patches React import statements in Objective-C files.
+ */
+export class ReactImportsPatchTransform extends Transform {
+  private chunkIndex: number = 0;
+  private readonly onlyTransformFirstChunk: boolean;
+  private readonly transformFn: (chunk: string) => string;
+
+  constructor(options?: Options) {
+    super();
+    this.onlyTransformFirstChunk = options?.onlyTransformFirstChunk ?? true;
+    this.transformFn = options?.transformFn ?? patchChunk;
+  }
+
+  _transform(chunk: any, _encoding: BufferEncoding, callback: TransformCallback): void {
+    const result =
+      this.onlyTransformFirstChunk && this.chunkIndex > 0
+        ? chunk.toString()
+        : this.transformFn(chunk.toString());
+    this.push(result);
+    ++this.chunkIndex;
+    callback();
+  }
+}
+
+/**
+ * Patch imports from a data chunk
+ * @param headerSet prebuilt React-Core header set
+ * @param chunk target chunk data
+ */
+export function patchChunk(chunk: string, headerSet: Set<string> = HEADER_SET): string {
+  let transformContent = chunk.replace(
+    /(?<=^\s*)#import\s+"(.+)"(?=\s*$)/gm,
+    (match: string, headerName: string): string => {
+      // `#import "RCTBridge.h"` -> `#import <React/RCTBridge.h>`
+      if (headerSet.has(headerName)) {
+        return `#import <React/${headerName}>`;
+      }
+
+      // `#import "React/RCTBridge.h"` -> `#import <React/RCTBridge.h>`
+      if (headerName.startsWith('React/')) {
+        const name = headerName.substring(6);
+        if (headerSet.has(name)) {
+          return `#import <React/${name}>`;
+        }
+      }
+
+      // Otherwise, return original import
+      return match;
+    }
+  );
+
+  transformContent = transformContent.replace(
+    /(?<=^\s*)#(if|elif)\s+__has_include\("(.+)"\)(?=\s*$)/gm,
+    (match: string, ifPrefix: string, headerName: string): string => {
+      // `#if __has_include("RCTBridge.h")` -> `#if __has_include(<React/RCTBridge.h>)`
+      if (headerSet.has(headerName)) {
+        return `#${ifPrefix} __has_include(<React/${headerName}>)`;
+      }
+
+      // `#if __has_include("React/RCTBridge.h")` -> `#if __has_include(<React/RCTBridge.h>)`
+      if (headerName.startsWith('React/')) {
+        const name = headerName.substring(6);
+        if (headerSet.has(name)) {
+          return `#${ifPrefix} __has_include(<React/${name}>)`;
+        }
+      }
+
+      // Otherwise, return original import
+      return match;
+    }
+  );
+
+  return transformContent;
+}
+
+// Fingerprint doesn't run `pod install` so we have a snapshot of the headers in the React-Core module from react-native 0.74.
+const HEADER_SET = new Set([
```

Comment:
> It doesn't seem to me like generateReactHeaderSetAsync needs pod install to have been run in order to get the header names? Maybe I'm missing something.

the `expo-modules-autolinking.ReactImportsPatcher` runs after `pod install`: https://github.com/expo/expo/blob/e456134de4e15fecfa27fad76c7280832c994db4/templates/expo-template-bare-minimum/ios/Podfile#L53

> Assuming I was missing something in the bullet point above and we do in fact need pod install, is there a way we can just look for all React/ header imports instead of needing a fixed list of headers (both here and in expo-modules-autolinking)? Or is there something special about public headers?

since we are trying to transform `#import "Header.h"` to `#import <React/Header.h>`, we cannot look for `React/` imports. i don't have a better solution than the hardcoded headers here. but i have other idea. let me answer in the other comment.